src/sonic-yang-models/yang-models/sonic-system-tacacs.yang

```diff
@@ -0,0 +1,172 @@
+module sonic-system-tacacs {
+    namespace "http://github.com/Azure/sonic-system-tacacs";
+    prefix ssys;
+    yang-version 1.1;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-port {
+        prefix port;
+    }
+
+    import sonic-portchannel {
+        prefix lag;
+    }
+
+    /*
+    import sonic-vlan {
+        prefix vlan;
+    }
+    */
+
+    import sonic-loopback-interface {
+        prefix loopback;
+    }
+
+    import sonic-interface {
+        prefix interface;
+    }
+
+//    import sonic-mgmt-port {
+//        prefix mgmt-port;
+//    }
+
+    revision 2021-04-15 {
+        description "Initial revision.";
+    }
+
+    typedef auth_type_enumeration {
+        type enumeration {
+            enum pap;
+            enum chap;
```

Comment:
Does SONiC support chap & mschap?

Reply:
As per the below HLD, chap & mschap are supported.
https://github.com/Azure/SONiC/blob/master/doc/aaa/TACACS%2B%20Authentication.md